src/main/java/com/minecolonies/coremod/colony/Colony.java

```diff
@@ -441,47 +506,115 @@ public int getID()
     }
 
     /**
-     * Increment the mobs killed by this colony.
-     * <p>
-     * Will award achievements for mobs killed.
+     * Get the amount of statistic.
+     *
+     * @param statistic the statistic.
+     * @return amount of statistic.
+     */
+    private int getStatisticAmount(@NotNull String statistic)
+    {
+        switch (statistic)
+        {
+            case TAG_GUARD_MOBS:
+                return killedMobs;
+            case TAG_MINER_ORES:
+                return minedOres;
+            case TAG_MINER_DIAMONDS:
+                return minedDiamonds;
+            case TAG_BUILDER_HUTS:
+                return builtHuts;
+            case TAG_FISHERMAN_FISH:
+                return caughtFish;
+            case TAG_FARMER_WHEAT:
+                return harvestedWheat;
+            case TAG_FARMER_POTATOES:
+                return harvestedPotatoes;
+            case TAG_FARMER_CARROTS:
+                return harvestedCarrots;
+            case TAG_LUMBERJACK_SAPLINGS:
+                return plantedSaplings;
+            case TAG_LUMBERJACK_TREES:
+                return felledTrees;
+            default:
+                return 0;
+        }
+    }
+
+    /**
+     * increment statistic amount.
+     * @param statistic the statistic.
+     */
+    private void incrementStatisticAmount(@NotNull String statistic)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') The Cyclomatic Complexity of this method "incrementStatisticAmount" is 12 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)